ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -23,6 +23,7 @@
 import io.confluent.ksql.execution.materialization.MaterializationInfo;
```

Comment:
I know I said that we should focus on getting everything working before worrying about code-cleanliness, and that the easiest way to get things working would be to duplicate SourceBuilder.java and add in the new logic, but now that you've got it working and are ready to open the PR, it's time to clean up the code :)

For example, we only bumped the execution step versions for TableSource and WindowedTableSource; StreamSource and WindowedStreamSource are exactly the same. Therefore, there's no point in having two identical code blocks in SourceBuilderV1 and SourceBuilder for building stream sources. We should de-dup, and this will also allow us to delete some of the duplicate tests in SourceBuilderTest and SourceBuilderV1Test.

As for how to de-dup the two builder classes, you could have SourceBuilder.java call helper methods from within SourceBuilderV1.java, or you could choose to have a separate base class that both inherit from (see StreamGroupByBuilderBase.java as an example). I'll let you take a look first to see what you think makes sense, though I'm also happy to provide more concrete suggestions if you'd like.